developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,311 @@
+* **sender** hex-coded account address of the sender of the transaction
+* **sequence_number** Sequence number of this transaction. This must match the sequence number stored in the sender's account at the time the transaction executes.
+*  **payload** The actual work that the Aptos blockchain needs to do.
+*  **max_gas_amount** Maximal total gas to spend for this transaction.
+*  **gas_unit_price** Price to be paid per gas unit.
+*  **expiration_timestamp_secs** If the current blockchain timestamp is greater than or equal to this time, then the transaction has expired and will be discarded.
+*  **chain_id** The chain id that this transaction is targeting to be ran on.
+
+### BCS
+Binary Canonical Serialization (BCS) is a serialization format. The design goals of BCS include:
+* provide good performance and concise (binary) representations
+* enforce canonical serialization, meaning that every value of a given type should have a single valid representation.
+* support a wide range of data types, including booleans, integers, strings, lists, maps and user-defined structs, etc. Note: floats, unicode chars and sets are not supported.
+
+BCS is not a self-describing format. As such, in order to deserialize a message, one must know the message type and layout ahead of time.
+
+An example of how BCS serializes a string.
+```typescript
+// A string is serialized as: byte length + byte representation of the string. The byte length is required for deserialization. Without it, no way the deserializer knows how many bytes to deserialize.
+const bytes: Unint8Array = bcs_serialize_string("aptos");
+assert(bytes == [0x61, 0x70, 0x74, 0x6F, 0x73]);
+```
+
+See [BCS][bcs] for more details.
+
+### Signing message
+A signing message is basically the bytes of the BCS serialized raw transaction. There is also a prefix, which is sha3_256("APTOS::RawTransaction"). Therefore, signing_message = prefix_bytes + bcs_bytes_of_raw_transaction.
+
+Below code shows how prefix bytes are calculated.
+```typescript
+function signinMessagePrefix(): Buffer {
+  let hash = SHA3.sha3_256.create();
+  hash.update(`APTOS::RawTransaction`);
+  return Buffer.from(hash.arrayBuffer());
+}
+```
+
+### Signature
+A signature is the [Ed25519][ed25519] encryption of a signing message with client private key. The signature is mainly used for security purpose.
+* By signing a signing message with private key, clients prove to the Aptos chain that they have authorized the transaction be executed.
+* Aptos chain will validate the signature with client account's public key to ensure that the transaction submitted is indeed signed by the client.
+* Signature is unique for each signing message/transaction. It is computationally expensive for hackers to calculate a signature out of the thin air without knowing private key. Moreover, because of sequence_number, a transaction cannot be replayed.
```

Comment:
probably better to discuss authenticators here and focus on the ed25519 signature

Reply:
👍